extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/common/GoogleErrorLogger.java

```diff
@@ -0,0 +1,26 @@
+package org.datatransferproject.datatransfer.google.common;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.UUID;
+import org.datatransferproject.spi.cloud.storage.JobStore;
+import org.datatransferproject.types.transfer.errors.ErrorDetail;
+
+/**
+ * Helper class that provides functionality for creating errors & logging them to data store.
+ * This class logs errors that should appear as failures in the completion email.
+ */
+public class GoogleErrorLogger {
+  public static void logFailedItemErrors(UUID jobId, ImmutableList<ErrorDetail> errorDetails, JobStore jobStore) throws IOException {
+    jobStore.addErrorsToJob(jobId, errorDetails);
+  }
+
+  public static ErrorDetail createErrorDetail(String idempotentId, String itemName, Exception e, boolean canSkip) {
```

Comment:
nit: use title instead of itemName. 

maybe also id instead of idempotentId.

Reply:
changed title

for idempotentId, I will keep it because it is a very important distinction, but not something that we can necessarily assert on. Also other classes use idempotentId, better to stick to convention in this case